src/Pages/Products/Products.js

```diff
@@ -14,37 +14,64 @@ import LineCardIconBlue from "../../assets/icons/LineCardIconBlue.svg";
 
 function Products(){
     const [brandID, setBrandID] = useState("All");
-    const {mode, setMode, cart, priceAfterDiscount} = useContext(ContextData);
+    const {allowed, mode, changeMode, cart, priceAfterDiscount} = useContext(ContextData);
     const [value, onChange] = useState(1000);
     const [sort, setSort] = useState(products);
+    const [page, setPage] = useState(1);
+    const listLenght = sort.filter(item=>(brandID === "All" && priceAfterDiscount(item.discount, item.originalPrice) < value ? item : brandID === item.categoryId && priceAfterDiscount(item.discount, item.originalPrice) < value)).length;
+    const lastPage = Math.ceil(listLenght/allowed);
+    const pageNumbers = [];
+    for(let i = 0; i < lastPage; i++){
+        pageNumbers.push(i+1);
+    }
     function hangleSorting(e){
         let option = e.target.value;
         if(option === "None"){
         setSort([...products].sort((a, b)=> {
             return a.id - b.id;
         }))
-        }else if(option === "Name"){
+        }else if(option === "Name A-Z"){
         setSort([...products].sort((a, b)=> {
             return a.title > b.title ? 1 : -1;
         }))
-        }else{
+        }else if(option === "Name Z-A"){
+            setSort([...products].sort((a, b)=> {
+                return b.title > a.title ? 1 : -1;
+            }))
+            }
+        else if(option === "Price min-max"){
+            setSort([...products].sort((a, b)=> {
+                return a.originalPrice - b.originalPrice;
+            }))
+            }
+        else{
         setSort([...products].sort((a, b) => {
-            return a.originalPrice - b.originalPrice;
+            return b.originalPrice - a.originalPrice;
         } ) )
```

Comment:
```
const handleSort=(option)=>{
const categorType=option.target.value
 const sortedProducts=products.filter(prod=>prod.type==categorType)
     .sort((a, b)=> a.id - b.id;
setSort(sortedProducts)
}
```

How about this one?
categoryType and prod.type need to matched, Please match object key to value

Reply:
It does not seem to be working

Reply:
let me check it after a while. If everything works, you can merge, we can refactor functions later.